lightly_studio/tests/resolvers/sample_resolver/test_sample_filter.py

```diff
@@ -0,0 +1,344 @@
+
+    def test_query__metadata_filter(self, test_db: Session) -> None:
+        dataset = create_dataset(session=test_db)
+        samples = create_images(
+            db_session=test_db,
+            dataset_id=dataset.dataset_id,
+            images=[
+                ImageStub(path="sample_0.png"),
+                ImageStub(path="sample_1.png"),
+            ],
+        )
+
+        # Create metadata
+        samples[0].sample["height"] = 100
+        samples[1].sample["height"] = 200
+
+        # Create the filter
+        sample_filter = SampleFilter(metadata_filters=[Metadata("height") > 150])
+
+        # Apply the filter
+        filtered_query = sample_filter.apply(query=select(SampleTable))
+        result = test_db.exec(filtered_query).all()
+
+        # Should return samples[1]
+        assert len(result) == 1
+        assert result[0].sample_id == samples[1].sample_id
+
+    def test_query__combination(self, test_db: Session) -> None:
+        dataset = create_dataset(session=test_db)
+        dataset_id = dataset.dataset_id
+        samples = create_images(
+            db_session=test_db,
+            dataset_id=dataset.dataset_id,
+            images=[
+                ImageStub(path="sample_0.png"),
+                ImageStub(path="sample_1.png"),
+                ImageStub(path="sample_2.png"),
+                ImageStub(path="sample_3.png"),
+            ],
+        )
+
+        # Sample ids for samples 0, 1, 2
+        sample_ids = [samples[0].sample_id, samples[1].sample_id, samples[2].sample_id]
+
+        # Tag samples
+        # Add tag1 to samples 1, 2 and tag2 to samples 0, 1
+        tag1 = create_tag(session=test_db, dataset_id=dataset_id, tag_name="tag1", kind="sample")
+        tag2 = create_tag(session=test_db, dataset_id=dataset_id, tag_name="tag2", kind="sample")
+        tag_resolver.add_sample_ids_to_tag_id(
+            session=test_db,
+            tag_id=tag1.tag_id,
+            sample_ids=[samples[1].sample_id, samples[2].sample_id],
+        )
+        tag_resolver.add_sample_ids_to_tag_id(
+            session=test_db,
+            tag_id=tag2.tag_id,
+            sample_ids=[samples[0].sample_id, samples[1].sample_id],
+        )
+
+        # Create metadata for samples 1, 2, 3
+        samples[1].sample["height"] = 100
+        samples[2].sample["height"] = 200
+        samples[3].sample["height"] = 300
+
+        # Create the filter
+        sample_filter = SampleFilter(
+            sample_ids=sample_ids,
+            annotation_label_ids=None,
+            tag_ids=[tag1.tag_id],
+            metadata_filters=[Metadata("height") < 250],
+        )
+
+        # Apply the filter
+        filtered_query = sample_filter.apply(query=select(SampleTable))
+        result = test_db.exec(filtered_query).all()
+
+        # Should return samples 1 and 2
+        assert len(result) == 2
+        assert {row.sample_id for row in result} == {samples[1].sample_id, samples[2].sample_id}
```

Comment:
We have a policy to avoid loops and iterators in tests. I’d suggest changing this. What do you think?